### PR TITLE
Send Email to User While register

### PR DESCRIPTION
**Pull request by SaniyaKumawat123**
Send Email to User While register

### CODE REVIEW

File: src/main/java/com/tripply/Auth/config/ApplicationConfiguration.java [hunk 1]
```diff
@@ -46,4 +47,13 @@ AuthenticationProvider authenticationProvider() {
 
         return authProvider;
     }
+
+    @Bean
+    public WebClient webClient() {
```
Comment (Jatin-Ongraph) [inline src/main/java/com/tripply/Auth/config/ApplicationConfiguration.java]: We don't need webClient bean, as It is already there in WebClientService class.

File: src/main/java/com/tripply/Auth/controller/UserController.java [hunk 2]
```diff
@@ -75,4 +75,13 @@ public ResponseEntity<ResponseModel<InvitationDetailResponse>> getInviteeDetails
         return ResponseEntity.ok(response);
     }
 
+
+    @GetMapping("/confirm/registration")
+    public ResponseEntity<ResponseModel<String>> confirmUserAccount(@RequestParam("inviteeEmail") String userEmail) {
```
Comment (Jatin-Ongraph) [inline src/main/java/com/tripply/Auth/controller/UserController.java:86]: Please make documentation on this API.

File: src/main/java/com/tripply/Auth/service/Impl/UserServiceImpl.java [hunk 3]
```diff
@@ -239,4 +243,39 @@ public ResponseModel<UserResponse> getUserById(UUID id) {
         return responseModel;
     }
 
+    @Override
+    public ResponseModel<String> updateUser(String userEmail) {
```
Comment (Jatin-Ongraph) [inline src/main/java/com/tripply/Auth/service/Impl/UserServiceImpl.java]: Please change method name to enableUser

File: src/main/java/com/tripply/Auth/service/Impl/AuthServiceImpl.java [hunk 4]
```diff
@@ -51,17 +51,23 @@ public ResponseModel<AuthenticationResponse> authenticateUser(LoginRequest login
...
+            response.setStatus(HttpStatus.OK);
 
-        log.info("AuthService: authenticateUser() ended with username -> {}", loginRequest.getEmail());
-        return response;
+            log.info("AuthService: authenticateUser() ended with username -> {}", loginRequest.getEmail());
+            return response;
+        } else {
+            response.setMessage("Please verify your Account");
+            response.setStatus(HttpStatus.UNAUTHORIZED);
+            return response;
```
Comment (Jatin-Ongraph) [inline src/main/java/com/tripply/Auth/service/Impl/AuthServiceImpl.java]: return response outside of if/else block.

File: src/main/java/com/tripply/Auth/service/Impl/UserServiceImpl.java [hunk 5]
```diff
@@ -239,4 +243,39 @@ public ResponseModel<UserResponse> getUserById(UUID id) {
...
+            response.setMessage("User not found");
+        }
+        return response;
+    }
+
+    private void sendEmailToUser(User user){
+        log.info("Begin sendMail() for the request: {} ", user.getFirstName());
+        try {
+            UserRequest userRequest = new UserRequest();
+            userRequest.setSendToName(user.getFirstName());
```
Comment (Jatin-Ongraph) [inline src/main/java/com/tripply/Auth/service/Impl/UserServiceImpl.java]: This should be fullName, use string.join

Comment (SaniyaKumawat123) [reply src/main/java/com/tripply/Auth/service/Impl/UserServiceImpl.java]: Yes, added

File: src/main/java/com/tripply/Auth/service/Impl/AuthServiceImpl.java (hunk 4 above)
Comment (SaniyaKumawat123) [reply src/main/java/com/tripply/Auth/service/Impl/AuthServiceImpl.java]: Updated

File: src/main/java/com/tripply/Auth/service/Impl/UserServiceImpl.java (hunk 3 above)
Comment (SaniyaKumawat123) [reply src/main/java/com/tripply/Auth/service/Impl/UserServiceImpl.java]: Done

File: src/main/java/com/tripply/Auth/controller/UserController.java (hunk 2 above)
Comment (SaniyaKumawat123) [reply src/main/java/com/tripply/Auth/controller/UserController.java:86]: Yes, done

File: src/main/java/com/tripply/Auth/config/ApplicationConfiguration.java (hunk 1 above)
Comment (SaniyaKumawat123) [reply src/main/java/com/tripply/Auth/config/ApplicationConfiguration.java]: Removed

**Review by Jatin-Ongraph — APPROVED** [review]

**Pull request merged**